presto-spark-base/src/main/java/com/facebook/presto/spark/execution/nativeprocess/HttpNativeExecutionTaskResultFetcher.java

```diff
@@ -185,7 +181,6 @@ private void onSuccess(PageBufferClient.PagesResponse pagesResponse)
         }
         token = nextToken;
         if (pagesResponse.isClientComplete()) {
-            completed = true;
             workerClient.abortResultsAsync(taskId);
             if (scheduledFuture != null) {
```

Comment:
this shall not be null at this point, right? could we force a non-null check here instead?